src/bandersnatch_storage_plugins/s3.py

```diff
@@ -0,0 +1,368 @@
+import configparser
+import contextlib
+import datetime
+import hashlib
+import logging
+import os
+import pathlib
+from pathlib import PurePath
+import tempfile
+from typing import (
+    IO,
+    Any,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Type,
+    Union,
+    TextIO
+)
+
+import boto3
+from botocore.client import Config
```

Comment:
I don't see both3 + botocore in requirements ...

Reply:
boto3 and botocore is brought by s3path , so I dont have to add it

Reply:
I strongly would suggest adding both boto3 + botocore in the requirements even though they technically may be redundant. 1) if for whatever reason s3path's dependencies change to not include either of these, bandersnatch is now broken (yay /s). 2) we do provide pinned (only top-level dependencies) requirement files in the repo and since bandersnatch directly depends on them they should have the same stability & tested-in-CI guarantees.